### PR TITLE
make sector access a relative path, allowing miners to move sector and metadata directories

### PR DESCRIPTION
**Pull request by laser**
Fixes [rust-fil-proofs #672](https://github.com/filecoin-project/rust-fil-proofs/issues/672).

## Why does this PR exist?

The existing implementation did not allow a storage miner to migrate storage or metadata directories from one path to some other.

## What's in this PR?

The staged and sealed sector metadata now include only the "sector access," which needs to be joined with the sector builder directory-values to create an absolute path.

### CODE REVIEW

**Comment by laser** [comment]
@dignifiedquire - This PR is (finally) ready for your review.

File: sector-builder-ffi/examples/simple.rs [hunk 1]
```diff
@@ -255,12 +258,25 @@ unsafe fn sector_builder_lifecycle(use_live_store: bool) -> Result<(), Box<Error
...
+    // migrate staged sectors, sealed sectors, and sector builder metadata to
+    // new directory (overwrites destination directory)
+    let renames = vec![
+        (metadata_dir_a.as_ref(), metadata_dir_b.as_ref()),
+        (staging_dir_a.as_ref(), staging_dir_b.as_ref()),
+        (sealed_dir_a.as_ref(), sealed_dir_b.as_ref()),
+    ];
+
+    for (from, to) in renames {
+        fs::rename(from, to).expect(&format!("could not rename from {:?} to {:?}", from, to));
```
Comment (dignifiedquire) [inline sector-builder-ffi/examples/simple.rs]: renames can easily fail, so shouldn't panic, but bubble up the error

File: sector-builder/src/disk_backed_storage.rs [hunk 2]
```diff
@@ -11,17 +11,29 @@ use crate::error::SectorManagerErr;
...
-// but this is not recommended, since some sealed sector sizes are invalid. If you must set this manually,
-// ensure the chosen sector size is a multiple of 32.
-
 pub struct DiskManager {
     staging_path: String,
     sealed_path: String,
 }
 
+impl DiskManager {
+    fn sector_path<P: AsRef<Path>>(&self, sector_dir: P, access: &str) -> PathBuf {
```
Comment (dignifiedquire) [inline sector-builder/src/disk_backed_storage.rs]: make this a standalone function, it doesn't use `self`

Comment (dignifiedquire) [inline sector-builder/src/disk_backed_storage.rs]: can be a standalone function, it doesn't use `self`

File: sector-builder-ffi/examples/simple.rs (hunk 1 above)
Comment (laser) [reply sector-builder-ffi/examples/simple.rs]: You are correct - but this is test code. If it fails, I _want_ it to panic and fail the test! :)

File: sector-builder/src/disk_backed_storage.rs (hunk 2 above)
Comment (laser) [reply sector-builder/src/disk_backed_storage.rs]: done

**Pull request merged**